php/nucleo/lib/toba_servicio_web_cliente_rest.php

```diff
@@ -104,11 +105,36 @@ static function get_datos_conexion($id_servicio)
 												'auth_password' => $conf[0][1], 
 												'to' => $conf[0][2], 
 												'auth_tipo' => 'basic'));
+            if (count($conf[9]) > 3) {
+                $datos['conexion']['version'] = $conf[0][3];
+            }
 		}
 		
 		//Devuelvo un ini para mantener contrato
 		$ini = new toba_ini();
 		$ini->set_entradas($datos);
 		return $ini;
 	}
+    
+    /**
+     * Devuelve la URL base con el nro de version incluido si existe
+     * @param array $opciones
+     * @return string
+     */
+    static function get_url_base($opciones): string
+    {        
+        if (! isset($opciones['version'])) {
+            return $opciones['to'];
+        }
+        
+        $valor = '';
+        $separadas = \explode('.', $opciones['version']);
+        if (! empty($separadas)) {
+            $valor = '/v'. \current($separadas);
+            $opciones['to'] = rtrim($opciones['to'], '/');
+        }
```

Comment:
En algún lugar tenemos que aclarar que la URL se va armar a partir de:

- valor de "to"
- caracter "/v"
- valor de "version"

En la forma <to><caracter><version>

Reply:
Algo [asi](https://github.com/SIU-Toba/framework/wiki/Envvar-Config#para-toba-34) va?

Reply:
Cual es el formato aceptado de versión?

- 1.0
- 1
- ambos?

Reply:
Ahi especifico el formato

Reply:
y que pasa si consumo la API de MP y es "1" solamente, no podrá especificarlo acá porque le hardcodeamos el valor "v", no?

Reply:
Si, eso seria un problema... el tema es que especificar todo en el campo podria generar un `pato gallareta`  en la url que no habria forma de atajar, si incluyo una validacion especifica de formato igual liquido a otras APIs que no se sirvan por el framework.

En definitiva la version a utilizar podria viajar como un header en lugar de ser parte de la url tambien, eso es particular de cada API no creo que podamos atajar todos los comportamientos de una forma segura.

Reply:
LOL pero ya teniamos eso, es decir... actualmente pueden poner lo que gusten en la URL, quizas simplemente hay que dejar el campo `version` como un extra que provee informacion pero no participa activamente del acceso al endpoint.

Otra opcion podria ser mover la invocacion del metodo dentro del `conectar` para determinar el parametro `to` el cual podes sobrescribir al momento de la invocacion.  Creo que esto podria ser simple y flexible.

Meter una callback seria raro porque esta parte esta pusheando la config que quiere usar si necesita una alternativa a la provista por el archivo ini.

Reply:
El callback era solo eso un raye o un posible escenario, como para entender el uso que le daría un módulo para customizarlo.

Que sea al momento de "conectar" y si es fácil de llevarlo hasta ese lugar y fácil de sobreescribirse por el módulo en cuestión, avancemos. De lo contrario podemos dejar sin efecto la intervención del valor "to"